src/avro_json_decoder.erl

```diff
@@ -49,21 +50,15 @@
 
 %%%_* APIs =====================================================================
 
-%% @doc Decode JSON schema.
--spec decode_schema(iodata()) -> avro_type().
-decode_schema(Json) ->
-  decode_schema(Json, fun(_) -> erlang:error(no_function) end).
+%% @doc Decode JSON format avro schema into erlavro internals.
+-spec decode_schema(binary()) -> avro_type().
+decode_schema(JSON) ->
+  parse_schema(decode_json(JSON)).
 
-%% @doc Decode Avro schema specified as Json string.
-%% Schema store or lookup function is needed to parse default values.
-%% @end
--spec decode_schema(iodata(), schema_store() | lkup_fun()) -> avro_type().
-decode_schema(JsonSchema, Store) when not is_function(Store) ->
-  Lkup = ?AVRO_SCHEMA_LOOKUP_FUN(Store),
-  decode_schema(JsonSchema, Lkup);
-decode_schema(JsonSchema, Lkup) ->
-  Decoded = decode_json(JsonSchema),
-  parse_schema(Decoded, Lkup).
+%% @doc Deprecated.
+-spec decode_schema(binary(), schema_store() | lkup_fun()) -> avro_type().
+decode_schema(JSON, _StoreOrLkup) ->
+  parse_schema(decode_json(JSON)).
```

Comment:
do we still need `?AVRO_SCHEMA_LOOKUP_FUN` macro in `avro_internal.hrl`?

Reply:
yes